store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java

```diff
@@ -231,8 +248,9 @@ protected void reset(Mailbox mbox, ZimbraSoapContext zsc) throws ServiceExceptio
     }
 
     private static void validateEmail(String email, Account account) throws ServiceException {
-        String[] addresses = account.getMailAddress();
-        if (Arrays.asList(addresses).contains(email)) {
+        String accountName = account.getName();
+        String aliases[] = account.getAliases();
+        if (Arrays.asList(aliases).contains(email) || accountName.equals(email)) {
             throw ServiceException
                 .FAILURE("Recovery address should not be same as primary email address.", null);
```

Comment:
How about having failure message as "Recovery address should not be same as primary/aliases email address."?
User might enter alias email and argue it is not same as primary email.